app/tasks.py

```
@@ -0,0 +1,122 @@
+# B2: Unbounded in-memory queue; grows without limit under load
+_queue: "asyncio.Queue[dict]" = asyncio.Queue()
+
+# B3: Retry on error by simply re-enqueuing the same item -> duplicate processing
+# B4: Naive de-dup heuristic based on rounded timestamp; collisions + misses
+_last_seen_ts_bucket: Dict[str, int] = {}  # event_id -> second bucket int
+
+# A3/A4: Background tasks are fire-and-forget; not tracked or cancelled on shutdown
+_background_tasks: list[asyncio.Task] = []
+
+
+async def process_event(event_id: str, payload: Optional[dict] = None) -> int:
```

Ruff (0.12.2)

Unused function argument: `payload`

(ARG001)